Alchemy pay : Buy boba  - Api setup

sk-enya opened the pull request:
Alchemy Pay
---
 - [x] Added custom domain mapping
 - [x] Added CORS on functions
 - [x] Added example env config file.

trangtran-enya commented on boba_examples/boba-custom-nft-bridge/contracts/L1NFTBridge.sol:
```diff
@@ -0,0 +1,365 @@
+    {
+        bytes memory extraData;
+        // if token has base on this layer
+        // if (pairNFTInfo[_l1Contract].baseNetwork == Network.L1) {
+        //     // check the existence of bridgeExtraData(uint256) on l1Contract
+        //     if (ERC165Checker.supportsInterface(_l1Contract, 0x9b9284f9)) {
+        //         extraData = iSupportBridgeExtraData(_l1Contract).bridgeExtraData(_tokenId);
+        //     } else {
+        //         // otherwise send tokenURI return (encoded in bytes)
+        //         // allow to fail if the call fails
+        //         extraData = abi.encode(IERC721Metadata(_l1Contract).tokenURI(_tokenId));
+        //     }
+        // }
```
Looks like it is not used. So I plan to remove this? @Boyuan-Chen can u please confirm if it's correct.

Boyuan-Chen replied:
Yup! @souradeep-das Please confirm!

souradeep-das replied:
yeah makes sense to remove them if we don't use them
but regarding the usage - I was thinking https://github.com/bobanetwork/boba/pull/332#issuecomment-1235825009

Boyuan-Chen replied:
They have already deployed on Avalanche.

souradeep-das replied:
ohh got it, then this is the only way. perfect, all good to remove the extraData part from the example then! 🚀

trangtran-enya replied:
Remove extraData

trangtran-enya commented on boba_examples/boba-custom-nft-bridge/contracts/L2NFTBridge.sol:
```diff
@@ -0,0 +1,441 @@
+        whenNotPaused()
+    {
+        bytes memory extraData;
+        // if token has base on this layer
+        // if (pairNFTInfo[_l2Contract].baseNetwork == Network.L2) {
+        //     // check the existence of bridgeExtraData(uint256) on l2Contract
+        //     if (ERC165Checker.supportsInterface(_l2Contract, 0x9b9284f9)) {
+        //         extraData = iSupportBridgeExtraData(_l2Contract).bridgeExtraData(_tokenId);
+        //     } else {
+        //         // otherwise send tokenURI return (encoded in bytes)
+        //         // allow to fail if the call fails
+        //         extraData = abi.encode(IERC721Metadata(_l2Contract).tokenURI(_tokenId));
+        //     }
+        // }
```
Looks like its not used so I plan to remove.

trangtran-enya commented on boba_examples/boba-custom-nft-bridge/contracts/L2NFTBridge.sol:
```diff
@@ -0,0 +1,441 @@
+        whenNotPaused()
+    {
+        bytes memory extraData;
+        // if token has base on this layer
+        // if (pairNFTInfo[_l2Contract].baseNetwork == Network.L2) {
+        //     // check the existence of bridgeExtraData(uint256) on l2Contract
+        //     if (ERC165Checker.supportsInterface(_l2Contract, 0x9b9284f9)) {
+        //         extraData = iSupportBridgeExtraData(_l2Contract).bridgeExtraData(_tokenId);
+        //     } else {
+        //         // otherwise send tokenURI return (encoded in bytes)
+        //         // allow to fail if the call fails
+        //         extraData = abi.encode(IERC721Metadata(_l2Contract).tokenURI(_tokenId));
+        //     }
+        // }
+        // size limits unchecked
```
same as above

trangtran-enya commented on boba_examples/boba-custom-nft-bridge/contracts/CustomERC721.sol:
```diff
@@ -0,0 +1,32 @@
+    function mint(address _to, uint256 _tokenId, bytes memory _data) external {
+        _safeMint(_to, _tokenId);
+
+        feature_1[_tokenId] = "1";
+        feature_2[_tokenId] = "2";
+        feature_3[_tokenId] = "3";
```
Why do we fix these values on L1? 🤔

Boyuan-Chen replied:
It’s an example. They said they have some skill values that are different from the standard ERC721.

trangtran-enya replied:
I mean this in constructor, I disallow creating customized features but always fixate on some values. If it's due to the ERC721 standard, I just wonder how other people handle if they want to customize these values.

Boyuan-Chen replied:
Could you please remove the unnecessary code? I will ask them!

trangtran-enya replied:
@Boyuan-Chen I have removed.

souradeep-das commented:
looking at the custom ERC721 here, and the features_1,2,3 requirements, looks like the current bridge already supports this functionality? The respective bridgeWithExtraData() method on the current bridge already allows to encode and move any kind of custom data along with the token, and use the data however they decide (although, it needs them to implement iSupportBridgeExtraData) - have they already deployed?

trangtran-enya commented:
> looking at the custom ERC721 here, and the features_1,2,3 requirements, looks like the current bridge could support this functionality? The respective bridgeWithExtraData() method on the current bridge would allow to encode and move any kind of custom data along with the token, and use the data however they decide (although, it needs them to implement iSupportBridgeExtraData) - is it that they have already deployed? in which case please disregard my comments https://github.com/bobanetwork/boba/tree/develop/boba_examples/nft_bridging#making-your-erc721-bridge-extra-data

I am not aware of  their needs in this case. Maybe we should ask them 🤔 ?

Boyuan-Chen commented:
They can upgrade their L1 NFTs to build `customData` payload and send to L2.

Closed.

sk-enya committed:
updated url + crypto change to BOBA

sk-enya committed:
change:
- Example env to publish for loading secret
- Update deploy.sh to export all env beofore deploy as deps to script

sk-enya committed:
change: mapping for custom domain creation

sk-enya committed:
changes:
- prepare deployment guide for serverless function
- setup loading of domain for cors and api deployment from env yml
- updated the serverless yml & example yml for env files

sk-enya committed:
changes:
- serverless split function not working
- Moved CORS origins to the custom section in serverless.yml as proper YAML arrays
- Updated the CORS configuration in the function to use the custom section:
- Removed CORS_ORIGINS from environment files since it's now handled in the serverless.yml directly.
- Updated the documentation to reflect these changes.

sk-enya committed:
changes:
- updated the serverless config
- updated documentation and gitignore & readme

sk-enya committed:
ignore .env* from git

sk-enya committed:
change: reverted unused code

sk-enya committed:
change updated variable naming

sk-enya committed:
Merge branch 'develop' into fix/alchemy-pay-boba-change

souhaillM reviewed (approved):
LGTM

sk-enya committed:
disable shellcheck
